src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -111,13 +123,49 @@ private static IComponentActivator GetComponentActivatorOrDefault(IServiceProvid
         /// </summary>
         protected internal ElementReferenceContext? ElementReferenceContext { get; protected set; }
 
+        internal HotReloadContext HotReloadContext { get; } = new();
+
+        private async void RenderRootComponentsOnHotReload()
+        {
+            await Dispatcher.InvokeAsync(() =>
+            {
+                if (_rootComponents is null)
+                {
+                    return;
+                }
+
+                HotReloadContext.IsHotReloading = true;
+                try
+                {
+                    foreach (var (componentState, initialParameters) in _rootComponents)
+                    {
+                        componentState.SetDirectParameters(initialParameters);
+                    }
+
+                    ProcessPendingRender();
```

Comment:
I don't think this line of code is actually needed, and won't do anything. Since we're not already in the context of a render batch, calling `SetDirectParameters` will start and execute a new renderbatch for each root component, and there will be no work left to do when we get to L145.

Could you remove this line if I'm right? Otherwise it may be confusing. I think it's perfectly fine to do a separate renderbatch for each root component here.